src/runtime/CollectionWrappers/IterableWrapper.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class IterableWrapper<T> : IEnumerable<T>
+    {
+        protected PyObject iterObject;
+        protected PyObject pyObject;
+
+        public IterableWrapper(PyObject pyObj)
+        {
+            pyObject = pyObj;
+            iterObject = new PyObject(Runtime.PyObject_GetIter(pyObj.Handle));
```

Comment:
`iterObject` should be local to `GetEnumerator` implementations, otherwise you won't be able to enumerate this instance twice.